t a/sw/source/core/layout/newfrm.cxx b/sw/source/core/layout/newfrm.cxx
index 86eefcb..7f19c1a 100644
--- a/sw/source/core/layout/newfrm.cxx
+++ b/sw/source/core/layout/newfrm.cxx
@@ -607,6 +607,7 @@ SwRootFrm::~SwRootFrm()
 {
     bTurboAllowed = sal_False;
     pTurbo = 0;
+    AllRemoveFtns();
     if(pBlink)
         pBlink->FrmDelete( this );
     if ( static_cast<SwFrmFmt*>(GetRegisteredInNonConst()) )

